nemo_curator/classifiers/content_type.py

```diff
@@ -0,0 +1,137 @@
+
+CONTENT_TYPE_IDENTIFIER = "nvidia/content-type-classifier-deberta"
+
+
+@dataclass
+class ContentTypeModelConfig:
+    model: str = "microsoft/deberta-v3-base"
+    fc_dropout: float = 0.2
+    max_len: int = 1024
+
+
+class ContentTypeModel(HFModel):
+    def __init__(
+        self,
+        config: ContentTypeModelConfig,
+        autocast: bool = False,
+        max_mem_gb: Optional[int] = None,
+    ):
+        self.config = config
+        self.autocast = autocast
+        if max_mem_gb is None:
+            max_mem_gb = _get_suggest_memory_for_classifier()
+
+        super().__init__(self.config.model, max_mem_gb=max_mem_gb)
+
+    def load_model(self, device: str = "cuda"):
+        model = HFDeberta.from_pretrained(CONTENT_TYPE_IDENTIFIER)
+        model.set_autocast(self.autocast)
+        model = model.to(device)
+        return model.eval()
+
+    def load_tokenizer(self):
+        return AutoTokenizer.from_pretrained(CONTENT_TYPE_IDENTIFIER)
+
+    def load_config(self):
+        return AutoConfig.from_pretrained(CONTENT_TYPE_IDENTIFIER)
+
+
+class ContentTypeClassifier(DistributedDataClassifier):
+    """
+    ContentTypeClassifier is a text classification model designed to categorize documents into one of 11 distinct speech types based on their content.
+    It analyzes and understands the nuances of textual information, enabling accurate classification across a diverse range of content types.
+    This class is optimized for running on multi-node, multi-GPU setups to enable fast and efficient inference on large datasets.
+
+    Attributes:
+        filter_by (list[str], optional): The classes to filter the dataset by.
+                                         If None, all classes will be included. Defaults to None.
+        batch_size (int): The number of samples per batch for inference. Defaults to 256.
+        text_field (str): The field in the dataset that should be classified.
+        pred_column (str): The column name where predictions will be stored. Defaults to "content_pred".
+        prob_column (str, optional): The column name where prediction probabilities will be stored. Defaults to None.
+        max_chars (int): The maximum number of characters in each document to consider for classification. Defaults to 2000.
+        device_type (str): The type of device to use for inference, either "cuda" or "cpu". Defaults to "cuda".
+        autocast (bool): Whether to use mixed precision for faster inference. Defaults to True.
+        max_mem_gb (int, optional): The maximum amount of memory in GB to allocate for the model. If None,
+                                      it defaults to the available GPU memory minus 4 GB.
+
+    """
+
+    def __init__(
+        self,
+        filter_by: Optional[List[str]] = None,
+        batch_size: int = 256,
+        text_field: str = "text",
+        pred_column: str = "content_pred",
+        prob_column: Optional[str] = None,
+        max_chars: int = 2000,
```

Comment:
We should probably bump this up to  to `5000` or something.  Max sequence length is 1024 for this model, assuming `4.5 chars` per token it is 4608 chars so hopefully that is good enough.